instrumentation/opensearch/opensearch-java-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opensearch/v3_0/QuerySplitter.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.opensearch.v3_0;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Splits multiple queries separated by semicolons. Single Responsibility: Only responsible for
+ * query separation logic.
+ */
```

Comment:
i dont think is right?

```suggestion
/**
 * Splits and joins queries for newline-delimited JSON (nd-json) format.
 * Splits input by newlines and joins output with semicolons for display.
 */
```